File: api/src/org/ufomatics/archipelago/control/UFOPlatformController.java
package org.ufomatics.archipelago.control;

import org.ufomatics.archipelago.model.*;
import org.ufomatics.archipelago.util.Localizer;

import java.util.Set;
import java.io.FileNotFoundException;
import java.io.IOException;

/**
 * @author Jérôme Beau
 * @version 7 mai 2006 16:56:21
 */
public interface UFOPlatformController {
    Localizer getLocalizer();

    Set<DatabaseAdapter> getAdapters();

    void close();

    Set<Database> getDatasources();

    MetaModel getMetaModel();

    boolean hasMetaModel();

    void loadMetaMapping(String filename) throws MetaException;

    void saveMetaMapping(String filename) throws IOException;

    MetaType getDefaultFieldType();

    DatabaseAdapter getDefaultAdapter();

    void addDatasource(Database database);

    Database createDatasource();

    Set<String> getRecentFiles();
}

File: src/archipelago/api/control/UFOPlatformController.ts
import {Database} from "archipelago/api/model/Database"
import {MetaModel} from "archipelago/api/model/MetaModel"
import {DatabaseAdapter} from "archipelago/api/model/DatabaseAdapter"
import {Localizer} from "archipelago/api/util/Localizer"
import {MetaType} from "archipelago/api/model/MetaType"
import {JSet} from "ts-jsdk"

/**
 *
 */
export interface UFOPlatformController {

  readonly localizer: Localizer

  adapters: DatabaseAdapter[]

  close(): void

  getDatasources(): Set<Database>

  getMetaModel(): MetaModel

  hasMetaModel(): boolean

  loadMetaMapping(filename: string): void;

  saveMetaMapping(filename: string): void;

  getDefaultFieldType(): MetaType

  getDefaultAdapter(): DatabaseAdapter | undefined

  addDatasource(database: Database): void

  createDatasource(): Database

  getRecentFiles(): JSet<String>
}
